### PR TITLE
Nonlinear A_s fix and rng fix for cell scores

### PR DESCRIPTION
**Pull request by justjhong**

### CODE REVIEW

File: src/scvi_v2/_module.py [hunk 1]
```diff
@@ -84,28 +84,29 @@ def setup(self):
...
         training = nn.merge_param("training", self.training, training)
-        u_drop = self.dropout(jax.lax.stop_gradient(u), deterministic=not training)
         sample_covariate = sample_covariate.astype(int).flatten()
         if not self.use_nonlinear:
+            u_drop = self.dropout(jax.lax.stop_gradient(u), deterministic=not training)
             A_s = self.A_s_enc(sample_covariate)
         else:
-            # A_s output by a non-linear function without an explicit intercept
+            # A_s output by a non-linear function without an explicit intercept.
+            u_drop = self.dropout(u, deterministic=not training)  # No stop gradient for nonlinear.
```
Comment (adamgayoso) [inline src/scvi_v2/_module.py:106]:
```suggestion
            u_drop = u  # No stop gradient for nonlinear.
```

Comment (justjhong) [reply src/scvi_v2/_module.py:106]: going to elave this since it is default 0 dropout
